### ISSUE TITLE
Have a sidebar with alternate routes

### ISSUE THREAD

**Issue by Makeshift** [Skretzo/shortest-path#120]
This is a rather large scope feature request, so I apologise in advance. 

I find myself changing the settings on this plugin very regularly to try and understand if a particular route would be easier with what I have in my inventory, or if I should run to the bank, or if I should use a permanent teleport instead of a consumable teleport, etc

It would be really cool to see a breakdown of how many tiles and what items are being used in alternate routes in the runelite sidebar, a bit like how Google Maps gives you a bunch of options for different public transport times.

For example, there could be 3 sections in the sidebar where the user can select different configurations. I would probably have one with `All (perm)`, one with `Inventory` and one with `Avoid wilderness` unticked.

**Comment by Skretzo** [comment]
**TL;DR:** Good idea. It's definitely possible, but maybe not as nice as Google Maps.

I like the idea, and it could probably be a nice addition to #48 which implements a drag-and-drop panel for picking the path you want to show (it is intended for different plugins competing to show a path, but should be fine to also let the Shortest Path plugin compete with itself over different config settings). I would probably not limit to 3 sections, but instead let you use the RuneLite profiles to make many different path alternatives. That way it can show the path for your other accounts (e.g. if you play both an ironman and a main account), though it wouldn't be 100% accurate as I cannot see quest progress on offline accounts. I believe RuneLite profiles can be custom made, so that would allow making the 3 alternatives you suggested without me having to recreate config selection in a side panel. It should be fairly easy to give a breakdown of number of tiles in the panel, but showing which items and teleports were used could be a UI nightmare. You should already get a decent breakdown from the world map tooltips that display number of tiles and which fairy ring codes were used, etc.. It could be problematic to show the statistics for many different path alternatives at the same time, because calculating many different paths might cause performance issues. I assume you want to know if alternative path 2 and 3 is almost as short as path 1? Calculating 3 paths and ranking them is probably doable, but I suspect that 10 or more paths would not be doable. I could maybe let the user click on each alternative to request to see it and then rank it, but doing 10 in advance would probably be a bad idea.

**Comment by Petrask24** [comment]
Could we save our own paths? As unlocks occur, the path could update the optimal order. For example:

Herb Runs:
Target 1: Varrock Patch
Target 2: Tree Gnome Stronghold
etc.

Sorry if this is is not the right place to request, it seems somewhat related. I found myself really wanting this feature for various tasks.
Thanks!
